packages/coil-extension/src/background/services/Stream.ts

```diff
@@ -197,19 +206,20 @@ export class Stream extends EventEmitter {
     this._debug('aborted because stream is no longer active.')
   }
 
-  async _getBtpToken() {
-    try {
-      this._debug('fetching btp token')
-      return await this._client.refreshBtpToken(this._token)
-    } catch (e) {
-      this._debug('error fetching btp token', e.message)
-      throw e
-    }
-  }
+  async _getBtpToken(): Promise<string> {
+    const cachedAnonToken = await this._anonTokens.getToken()
+    if (cachedAnonToken) return cachedAnonToken
 
-  async _makePlugin() {
-    const btpToken = await this._getBtpToken()
+    await this._anonTokens.populateTokens(
```

Comment:
we might want to populate tokens early, like somehow returning a warning from `getToken` if you're running low. Then we don't need to put a costly populateTokens call on the critical path to web monetizing a page

We should also probably have some kind of lock in the token service to prevent two populate calls from running at once. If I switch between WM tabs while populateTokens is running then it might start another populateTokens call and get me too many tokens